### PR TITLE
WIP: Refactor Patternfly Notification drawer

### PR DESCRIPTION
**Pull request by glekner**
Refactoring the Notification drawer with the PF-React implementation.
In order to run foreman with the PF-React drawer, please clone my [branch](https://github.com/gilad215/patternfly-react/tree/feature-notifdrawer) and run `npm link`. then run `npm link patternfly-react` in foreman

**Works**

-  Notifications are fetched and displayed correctly
-  `togglePanel`, `onNotificationClick`, `onMarkPanelAsRead`, `toggleDrawerHide`

**Doesn't Work (Needs to be implemented in Foreman)**
- `onMarkPanelAsClear`, `onClickedLink`, `onNotificationHide`
- CSS is being used from the inner flie `notifications.scss`, I patched it a bit and it works partially with my drawer but I think we should fetch the CSS from Patternfly

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -53,6 +53,7 @@ gem 'sshkey', '~> 1.9'
 gem 'dynflow', '>= 0.8.29', '< 1.0.0'
 gem 'daemons'
 gem 'get_process_mem'
+gem 'rdoc'
```
Comment (ohadlevy) [inline Gemfile]: not related / required?

File: bundler.d/journald.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # disable to avoid journald native gem in development setup
 group :journald do
-  gem 'logging-journald', '~> 1.0'
+ # gem 'logging-journald', '~> 1.0'
```
[the remark this answers is not in the document]
Comment (ohadlevy) [reply bundler.d/journald.rb]: you can use ```bundle install --without journald``` to get the same effect.

File: webpack/assets/javascripts/react_app/components/notifications/index.js [hunk 3]
```diff
@@ -38,19 +39,26 @@ class notificationContainer extends React.Component {
...
-            <Drawer
-              onExpandGroup={expandGroup}
-              onClickedLink={onClickedLink}
-              onMarkAsRead={onMarkAsRead}
-              onMarkGroupAsRead={onMarkGroupAsRead}
-              expandedGroup={expandedGroup}
-              notificationGroups={notifications}
-              toggleDrawer={toggleDrawer}
+            <NotificationDrawerWrapper
+            panels={formatedPanels}
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/components/notifications/index.js]: nitpick: wrong indentation.

**Review by ohadlevy — CHANGES_REQUESTED** [review]
thanks @gilad215, a few code review comments inline.

regarding the missing features:
onMarkPanelAsClear - I assume this should be a new API call to bulk delete actual notifications.

onClickedLink - Why can't we handle links? not sure I follow?

onNotificationHide - is this hiding the notification (same as clear?) or closing the drawer? I assume on click outside works as before?

File: Gemfile (hunk 1 above)
Comment (glekner) [reply Gemfile]: Foreman won't compile at my end without this line, you can ignore this because ill be working on my Mac soon and I will delete this.

**Comment by ohadlevy** [comment]
I'm not sure whats the issue TBH: but at worse you can either not include
this file, or better add it to your bundle.d/Gemfile.local.rb instead.

**Comment by glekner** [comment]
@ohadlevy 
`onClickedLink` - I think i can fix this at my branch, ill take a look.
`onNotificationHide` - This is the button in the dropdown that says "Hide this notification". It was my idea to add this so its not mandatory. we should treat this the same as `onMarkPanelAsClear` just deleting a single notification by its key instead of the whole panel.

**Comment by glekner** [comment]
- Fixed `onClickedLink`
- Added more CSS to `notifications.scss` to enable in-panel scrolling in scenarios where one panel has 5+ notifications

**Comment by glekner** [comment]
Added `onNotificationHide`, `onMarkPanelAsClear` functionality.

TODO: Tests

**Comment by glekner** [comment]
- Added New + Missing tests for the Notification Reducers/Actions
- Waiting for the drawer merge on PF-React, any suggestions for the codeclimate tests? @sharvit

**Pull request closed without merging**